easybuild/easyblocks/generic/buildenv.py

```diff
@@ -28,14 +28,35 @@
 
 @author: Alan O'Cais (Juelich Supercomputing Centre)
 """
+import os
+
 from easybuild.easyblocks.generic.bundle import Bundle
+from easybuild.tools.config import build_option, update_build_option
+from easybuild.tools.toolchain.toolchain import RPATH_WRAPPERS_SUBDIR
 
 
 class BuildEnv(Bundle):
     """
     Build environment of toolchain: only generate module file
     """
 
+    def prepare_step(self, *args, **kwargs):
+        """
+        Custom prepare step for buildenv: export rpath wrappers if they are being used
+        """
+        # We export the rpath wrappers under special conditions
+        # (the wrappers are needed if LD_LIBRARY_PATH is being filtered)
+        filtered_env_vars = build_option('filter_env_vars') or []
+        if build_option('rpath') and 'LD_LIBRARY_PATH' in filtered_env_vars and 'LIBRARY_PATH' not in filtered_env_vars:
+            # re-create installation dir (deletes old installation),
+            self.make_installdir()
+            # then set keeppreviousinstall to True (to avoid deleting wrappers we create)
+            self.cfg['keeppreviousinstall'] = True
+            # export our RPATH wrappers during the prepare step
+            update_build_option('rpath_wrappers_dir', os.path.join(self.installdir, 'bin'))
```

Comment:
I don't think this is the right approach...

For one, this applies to the whole EasyBuild session, so if other installations are done afterwards, they'll all "inherit" the `rpath_wrapper_dirs` setting, which is not what's intended.

This is another sign for me that `rpath_wrappers_dir` should be an easyconfig parameter, not a general EasyBuild configuration option.